core/trashcan.js

```diff
@@ -96,12 +109,46 @@ Blockly.Trashcan.prototype.SPRITE_LEFT_ = 0;
  */
 Blockly.Trashcan.prototype.SPRITE_TOP_ = 32;
 
+/**
+ * The openness of the lid when the trashcan contains blocks.
+ *    (0.0 = closed, 1.0 = open)
+ * @type {number}
+ * @private
+ */
+Blockly.Trashcan.prototype.HAS_BLOCKS_LID_ANGLE = 0.1;
+
+/**
+ * The maximum number of blocks that can go in the trashcan's flyout. '0' turns
+ * turns off trashcan contents, 'Infinity' sets it to unlimited.
+ * @type {number}
+ */
+Blockly.Trashcan.prototype.MAX_CONTENTS = 32;
+
 /**
  * Current open/close state of the lid.
  * @type {boolean}
  */
 Blockly.Trashcan.prototype.isOpen = false;
 
+/**
+ * The minimum openness of the lid. Used to indicate if the trashcan contains
+ *  blocks.
+ * @type {number}
+ */
+Blockly.Trashcan.prototype.minOpenness_ = 0;
+
+/**
+ * True if the trashcan contains blocks, otherwise false.
+ * @type {boolean}
+ */
+Blockly.Trashcan.prototype.hasBlocks = false;
+
+/**
+ * A list of Xml representing blocks "inside" the trashcan.
+ * @type {Array}
```

Comment:
This and `hasBlocks` can be private.  `private` in closure annotations [means](https://github.com/google/closure-compiler/wiki/Annotating-JavaScript-for-the-Closure-Compiler#private) it can only be used in the same file, so your use of these in the delete handler will be fine.

Reply:
Yes.  If anyone wants access later, we can add a getter function.